File: sdk/java-guest/examples/http.java
package dev.latent.app;
import dev.latent.generated.Bindings;
import dev.latent.generated.Bindings.*;
import dev.latent.guest.*;
import java.nio.charset.StandardCharsets;
import java.util.List;

public final class Capsule implements Bindings.Exports {
    public Unsigned64 run(Long which, String text, Unsigned64 handle) {
        var method = which == 0 ? LatentHttpClientMethod.Get : which == 1 ? LatentHttpClientMethod.Head : LatentHttpClientMethod.Post;
        var result = LatentHttpClient.send(new LatentHttpClientRequest(method, text, List.of(),
            Option.some("payload".getBytes(StandardCharsets.UTF_8)), Option.some("text/plain"),
            Option.none(), Option.some(new Unsigned64(1000))));
        if (!result.isError()) return new Unsigned64(result.value().status() + 1000L * result.value().body().length);
        return new Unsigned64(switch (result.error().tag()) {
            case 2 -> 10; case 12 -> 11; default -> throw new IllegalStateException("unexpected HTTP error");
        });
    }
}

File: sdk/dotnet-guest/examples/http.cs
using System.Text;
using Lsf.Guest;
using Raw = ServiceWorld.wit.Imports.latent.http.IClientImports;
namespace ServiceWorld.wit.Exports.tests.http;
public class ApiExportsImpl : IApiExports
{
    public static ulong Run(uint which, string text, ulong handle)
    {
        var result = Http.Send(new Raw.Request(which == 0 ? Raw.Method.GET : which == 1 ? Raw.Method.HEAD : Raw.Method.POST,
            text, [], Encoding.UTF8.GetBytes("payload"),
            "text/plain", null, 1000));
        if (!result.IsOk) return result.AsErr.Tag switch {
            Raw.HttpError.Tags.PermissionDenied => 10,
            Raw.HttpError.Tags.Uncertain => 11,
            _ => throw new System.InvalidOperationException("unexpected HTTP result") };
        var value = result.AsOk;
        return value.status + 1000UL * (ulong)(value.body?.Length ?? 0);
    }
}
